docs/blog/2024-09-05-personal-access-tokens.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Better security with Personal Access Tokens
+slug: personal-access-tokens
+authors: ['Philip Fulcher']
+tags: [nx-cloud]
+cover_image: /blog/images/2024-09-05/personal-access-tokens-header.avif
+---
+
+Today, Nx Cloud gets a huge upgrade to managing access to your cached artifacts
+using [Nx Replay](/ci/features/remote-cache). Previously, workspaces
+were limited to defining access tokens with read or read/write permissions for an entire workspace. With the
+introduction of _personal access tokens_, you gain much more control over access. This is a feature request we've heard
+from many customers, especially our [Enterprise](/enterprise) customers, and we're happy to be able to deliver this
+enhancement.
+
+## Access Tokens and the problem of revoking access
+
+Our previous implementation of access tokens required you to commit the access token to your `nx.json` file. Typically,
+service providers don't recommend committing any kind of API token like this, and we've fielded a lot of questions about
+this practice in the past. It is safe to commit this token, as access to the cached artifacts of Nx Cloud rely on both
+the access token and the source code itself. Without access to both, you can't access the cache.
+
+However, this did present the following problem: revoking someone's access to the cache became difficult as long as they
+had the source code. Imagine a scenario where someone has left an organization. As long as they have a clone of the
+repo, they have everything they need to access the cache, even if their credentials have been
+revoked. To fully revoke
+their access would require cycling the access token, which could interrupt the work of other developers and CI
+pipelines.
+
+## Access Tokens become CI Access Tokens
+
+What we previously called "access tokens" will now be called "[CI access tokens](/ci/recipes/security/access-tokens)."
+They are still defined at the workspace
+level, but are designed for use in CI. These tokens can now be set as environment variables on your CI platform so that
+they're no longer committed to your repo.
+
+## What are personal access tokens?
+
+[Personal access tokens](/ci/recipes/security/personal-access-tokens) are a new type of access token that is scoped to
+an individual user. This means that this token lives and dies with that member's access to your Nx Cloud workspace.
+Users must log in to Nx Cloud and confirm their access to a workspace before a personal access token can be created.
+Once created, we validate that token for access each time. As soon as a user loses access, the personal access token no
```

Comment:
```suggestion
Once created, we validate that token for access each time you use nx-cloud locally on your machine. As soon as a user loses access, the personal access token no
```

Don't really have strong opinions other than the fact that we should be more specific than `each time`. Each time of what?